dask_expr/_expr.py

```diff
@@ -3752,34 +3765,20 @@ def _broadcast_dep(self, dep: Expr):
         # Always broadcast single-partition dependencies in Fused
         return dep.npartitions == 1
 
-    def _task(self, index):
-        graph = {self._name: (self.exprs[0]._name, index)}
+    def _task(self, name: Key, index: int) -> Task:
+        subgraphs = {}
         for _expr in self.exprs:
-            if isinstance(_expr, Fused):
-                subgraph, name = _expr._task(index)[1:3]
-                graph.update(subgraph)
-                graph[(name, index)] = name
-            elif self._broadcast_dep(_expr):
-                # When _expr is being broadcasted, we only
-                # want to define a fused task for index 0
-                graph[(_expr._name, 0)] = _expr._task(0)
+            if self._broadcast_dep(_expr):
+                subname = (_expr._name, 0)
             else:
-                graph[(_expr._name, index)] = _expr._task(index)
+                subname = (_expr._name, index)
+            subgraphs[subname] = _expr._task(subname, subname[1])
 
         for i, dep in enumerate(self.dependencies()):
-            graph[self._blockwise_arg(dep, index)] = "_" + str(i)
+            subgraphs[self._blockwise_arg(dep, index)] = "_" + str(i)
```

Comment:
I have to admit that I don't understand what's going on here but I just went ahead and translated the existing code